internal/cmd/versions_test.go

```diff
@@ -12,30 +12,184 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package cmd_test
+package cmd
 
 import (
-	"fmt"
+	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 
+	"github.com/tetratelabs/getenvoy/internal/test/morerequire"
 	"github.com/tetratelabs/getenvoy/internal/version"
 )
 
-func TestGetEnvoyVersions(t *testing.T) {
-	o, cleanup := setupTest(t)
-	defer cleanup()
-
-	// Run "getenvoy versions"
-	c, stdout, stderr := newApp(o)
-	o.Out = stdout
-	err := c.Run([]string{"getenvoy", "versions"})
-
-	// Verify the command invoked, passing the correct default commandline
-	require.NoError(t, err)
-	require.Equal(t, fmt.Sprintf(`VERSION	RELEASE_DATE
-%s	2020-12-31
-`, version.LastKnownEnvoy), stdout.String())
-	require.Empty(t, stderr)
+func TestGetInstalledVersions(t *testing.T) {
```

Comment:
I find this test confusing for a couple if reasons:

usually t.Run is used whenever you have a test matrix. It is not enforced but it is very convenient. In this case you don't have that but still run different subtests inside the same tests. This make think those tests are coupled. How about moving the tests into a test matrix and the asumptions being executed in a function?

Something like 
```
type testCase struct{
   setup func() // initialize conditions for tests
   homeDir string
   expectedErr error
   expectedRows Rows
   tearDown func() // remove whatever you created
}
```

Reply:
doing it like this will seem more or similar work than creating separate test functions... having to have a 5 part struct, which ignores when there should be no error (means you have to check to see if that is nil or not) is more digging a hole.

I don't really agree with some go conventions frankly, they are worse if this is the solution

Reply:
when you look at tests forced into this pattern, the run block is a mess that looks nothing like normal code. it is if tc.X == "" { assert true } else {assert equals} etc etc. it is a mess. I don't understand why people do this except cargo cult or someone's blog, but the result is almost always a mess.

Reply:
I went the route of breaking up the test as it is more readable than this or forcing all the tests into the same matrix. The only thing less readable is the test output as we have to use camel case instead of sentences.

Reply:
Yeah, probably making it fit into the same matrix isn't the best idea given your point that assertions are made at different points (in my case I saw all the assertions were around error and rows but yeah they happened at different stages). Splitting the test looks good.